### ISSUE TITLE
GeneratePackageOnBuild Not Including Assets added to project by previous build step.

### ISSUE THREAD

**Issue by groogiam** [dotnet/sdk#28297]
### Describe the bug
I have a Razor Class Library with GeneratePackageOnBuild set. This library has a dependency on another library that copies its output (a large amount of files to its wwwroot directory using a Target with `AfterTargets="Build"` with the following custom task.

```msbuild
	<UsingTask TaskName="RenameDirectory" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
		<ParameterGroup>
			<DirectoryName ParameterType="System.String" Required="true" />
			<NewDirectoryName ParameterType="System.String" Required="true" />
		</ParameterGroup>
		<Task>
			<Using Namespace="System" />
			<Using Namespace="System.IO" />
			<Using Namespace="System.Text.RegularExpressions" />
			<Code Type="Fragment" Language="cs">
				<![CDATA[
          if(Directory.Exists(NewDirectoryName))
            Directory.Delete(NewDirectoryName, true);
          Directory.Move(DirectoryName, NewDirectoryName);
        ]]>
			</Code>
		</Task>
	</UsingTask>
```

When doing this the Generated nuget package does not contain the dynamically generated assets on the initial build or after a clean is performed even though the solution has a dependency set on the first project. The msbuild output seems to indicated that the Blazor class library build is not started until after the first library is completed, but the output is not in the nuget package.

Oddly enough this does not seem to happen when building inside Visual Studio. It only happens when using dotnet build on the solution.

### To Reproduce
Working on a simple repro. This only seems to happen when the post build operation is done on a large directory which I'm having trouble replicating in a simple repo. 

We may close this issue if:
- the repro project you share with us is complex. We can't investigate custom projects, so don't point us to such, please.
- if we will not be able to repro the behavior you're reporting
-->

### Exceptions (if any)
No exceptions are encountered.

### Further technical details
```
C:\Users\Greg.Horvath>dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.401
 Commit:    0906eae6f8

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.401\

global.json file:
  Not found

Host:
  Version:      6.0.9
  Architecture: x64
  Commit:       163a63591c

.NET SDKs installed:
  6.0.304 [C:\Program Files\dotnet\sdk]
  6.0.401 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info
```
Visual Studio 17.3.5

**Comment by groogiam** [comment]
After looking at some bin logs this appears to be due to parallel builds. When building in parallel the dependent project gets loaded before the output is generated. The pack files seem to be determined at project load. I'm not sure if there is any way to prevent this. I'm trying to add some solution dependencies but no matter what I do the dependent solution seems to get loaded in a parallel thread. Using `dotnet msbuild m:1` seems to work but it would be nice to have this working in a parallel build.

On a side note there does appear to be some incorrect documentation as the dotnet build is supposed to be a https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-command-line-reference?view=vs-2022 maxCpuCount says that this is defaulted to 1 if not specified. 

While [here ](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build#msbuild) it says "Running dotnet build is equivalent to running dotnet msbuild -restore; however, the default verbosity of the output is different." This would leave me to believe that the dotnet build would be not be parallel but it appears that it is.

**Comment by marcpopMSFT** [comment]
Old issue triage: it sounds like you figured out the parallel build issue so closing.
